app/src/main/java/com/github/geohunt/app/ui/components/statistics/DateGraph.kt

```diff
@@ -0,0 +1,65 @@
+package com.github.geohunt.app.ui.components.statistics
+
+import androidx.compose.runtime.Composable
+import com.github.geohunt.app.i18n.DateFormatUtils
+import com.github.geohunt.app.ui.components.utils.Graph
+import java.time.LocalDate
+import java.time.temporal.ChronoUnit
+
+/**
+ * A graph that draws x/y points, where x is a date and y is a value
+ * @param xDateValues The x values of our points represented by dates
+ *  have to be in the range dateGranularity.dateRange(LocalDate.now())
+ * @param dateGranularity the DateGranularity of the graph, gives the range of dates that will
+ *  be shown on the graph (last week, month, year)
+ * @param yValues the y values of our points represented by Long values
+ */
+@Composable
+fun DateGraph(
+        xDateValues: List<LocalDate>,
+        dateGranularity: DateGranularity,
+        yValues: List<Long>,
+) {
+    require(xDateValues.isNotEmpty())
+    require(yValues.isNotEmpty())
+    require(xDateValues.size == yValues.size)
```

Comment:
The require() functions are a good way to validate that the input parameters match our criteria. The only problem is that require() function throws IllegalArgumentException. which may not always be descriptive enough.

I highly suggest replacing require() by check(), which you could use for example in the following way:
`check(xDateValues.isNotEmpty()) { "xDateValues cannot be empty" }`.
This way if the xDateValues would be empty, we would throw `IllegalStateException("xDateValues cannot be empty")`, which would be much more informative in case of a debugging!

Reply:
Using require makes more sense since we are looking at the given arguments. But I agree that I should have added a simple debugging message to make it more clear.